File: nexus/star.py
"""Auto-star Nexus Memory System repository."""
import os
import subprocess
from pathlib import Path
from threading import Thread


def star_repo_background():
    """Star the repository in the background if not already attempted."""
    if os.environ.get("NEXUS_NO_STAR") == "1":
        return

    marker = _marker_path()
    if marker.exists():
        return

    Thread(target=_star_repo_impl, daemon=True).start()


def _marker_path() -> Path:
    """Get the path to the star attempt marker file."""
    config_home = os.environ.get("XDG_CONFIG_HOME")
    if config_home:
        config_dir = Path(config_home)
    else:
        config_dir = Path.home() / ".config"
    return config_dir / "nexus-memory-system" / ".star-attempted"


def _star_repo_impl():
    """Attempt to star the repository using gh CLI."""
    marker = _marker_path()
    marker.parent.mkdir(parents=True, exist_ok=True)

    try:
        subprocess.run(
            ["gh", "auth", "status"],
            capture_output=True,
            check=True,
            timeout=2,
        )
    except (subprocess.CalledProcessError, FileNotFoundError, subprocess.TimeoutExpired):
        marker.touch()
        return

    try:
        subprocess.run(
            ["gh", "api", "--silent", "-X", "PUT", "/user/starred/scooter-lacroix/Nexus-Memory-System"],
            capture_output=True,
            timeout=5,
        )
    except (subprocess.CalledProcessError, FileNotFoundError, subprocess.TimeoutExpired):
        pass

    marker.touch()

File: crates/nexus-cli/src/star.rs
use std::path::PathBuf;
use std::process::Command;

pub fn star_repo_background() {
    if std::env::var("NEXUS_NO_STAR").unwrap_or_default() == "1" {
        return;
    }

    let marker = marker_path();
    if marker.exists() {
        return;
    }

    std::thread::spawn(move || {
        let _ = star_repo_impl();
    });
}

fn marker_path() -> PathBuf {
    let config_dir = std::env::var("XDG_CONFIG_HOME")
        .map(PathBuf::from)
        .unwrap_or_else(|_| {
            let home = std::env::var("HOME").unwrap_or_else(|_| ".".to_string());
            PathBuf::from(home).join(".config")
        });
    config_dir.join("nexus-memory-system").join(".star-attempted")
}

fn star_repo_impl() -> std::io::Result<()> {
    let marker = marker_path();
    if let Some(parent) = marker.parent() {
        std::fs::create_dir_all(parent)?;
    }

    if Command::new("gh").arg("auth").arg("status").output().is_err() {
        std::fs::write(&marker, "")?;
        return Ok(());
    }

    let _ = Command::new("gh")
        .args(["api", "--silent", "-X", "PUT", "/user/starred/scooter-lacroix/Nexus-Memory-System"])
        .output();

    std::fs::write(&marker, "")?;
    Ok(())
}
